app/modules/enduser_setup.c

```diff
@@ -1109,7 +1233,7 @@ static void enduser_setup_dns_recv_callback(void *arg, char *recv_data, unsigned
 
 static void enduser_setup_free(void)
```

Comment:
Can this be called while a timer is still running (say the 10 second finish timer)? If so, then there is a problem as the timer structure (which is part of the state structure) is still being referenced by the timer subsystem. Might be safer to explicitly cancel any outstanding timers.